server/webdriver_tests/tests/place_explorer_test.py

```diff
@@ -186,3 +186,82 @@ def test_demographics_redirect_link(self):
         # Assert chart title is correct.
         self.assertEqual("Median age by gender: states near California(2020)",
                          chart_title)
+
+    def test_ranking_chart_present(self):
+        """
+        Test basic ranking chart.
+        """
+        CHART_TITLE = "Median individual income: rankings in United States of America(2020)"
+        CHART_TABLE_HIGHEST_TITLE = "Highest"
+        CHART_TABLE_LOWEST_TITLE = "Lowest"
+        CA_ECONOMICS_URL = CA_URL + "?category=Economics"
+        FIRST_RANK = "1."
+        LAST_RANK = "52."
+
+        self.driver.get(self.url_ + CA_ECONOMICS_URL)
+
+        chart_title_present = EC.text_to_be_present_in_element(
+            (By.XPATH, '//*[@id="main-pane"]/section[5]/div/div[5]/div/h4'),
+            CHART_TITLE)
+        WebDriverWait(self.driver, self.TIMEOUT_SEC).until(chart_title_present)
+
+        chart_subtitle_present = EC.presence_of_element_located(
+            (By.XPATH,
+             '//*[@id="main-pane"]/section[5]/div/div[5]/div/div/div/h4'))
+        WebDriverWait(self.driver,
+                      self.TIMEOUT_SEC).until(chart_subtitle_present)
+
+        chart_subtitle = self.driver.find_element_by_xpath(
+            '//*[@id="main-pane"]/section[5]/div/div[5]/div/div/div/h4').text
+        self.assertTrue = ("California ranks" in chart_subtitle)
+
+        chart_table_highest = self.driver.find_element_by_xpath(
+            '//*[@id="main-pane"]/section[5]/div/div[5]/div/div/div/div/div[1]')
+        chart_table_lowest = self.driver.find_element_by_xpath(
+            '//*[@id="main-pane"]/section[5]/div/div[5]/div/div/div/div/div[2]')
+
+        chart_table_highest_title = EC.text_to_be_present_in_element((
```

Comment:
I think instead of waiting for the text to be present in the element, it'll be better to wait for the element to be present, then assert that the text in the element is expected. This way, we know if a test fails because an element doesn't show up or if the text is incorrect. Here and chart_table_lowest_title and checking the redirect link in the next test

Reply:
Done